templates/zerver/create_realm.html

```diff
@@ -4,10 +4,29 @@
 {# This is where we pitch the app and solicit signups. #}
 
 {% block portico_content %}
+<script src='https://www.google.com/recaptcha/api.js'></script>
 <script type="text/javascript">
 $(function () {
     common.autofocus('#email');
+    var submitButton = $("#send_confirm").find(":submit");
+    submitButton.click(validate);
 });
+
+function onSubmit () {
+    var response = grecaptcha.getResponse();
+    $('#g-recaptcha-response').text(response);
+    document.getElementById('send_confirm').submit();
+}
+
+function validate (event) {
+    var form = document.getElementById('send_confirm');
+    if (form.checkValidity()) {
```

Comment:
It's not obvious to me that we need this step. Can you explain what it's doing? I could definitely be missing something.

If we don't need it, let's do this the simple way -- following the example right at the top of https://developers.google.com/recaptcha/docs/invisible .

Reply:
This just keeps our current form validation intact. Currently, if a user submits this form without entering the email, we don't send the request to the server. We show an error message. If we remove this client side validation, everything will still work but the form validation will be done by Django. You want me to add a comment for this?